### DISCUSSION TITLE
OPENRADIOSS PARALELLING

### DISCUSSION THREAD

**Discussion by CinekPoland** [OpenRadioss/OpenRadioss#943] [Show and tell]
I did some simulations of bird strike https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/8847361/Bird+Strike+On+Windshield
 using 2 , 8, 16 and also 32 cores.  scalability  wasn't very good.  Using 16 cores I achieved about 4.5 speedup są it is not satisfactory.  Doeas anyone know it is possible to improve scalability ?  Maybe scalability could be better if simulations will be bigger ? Doeas anyone knows where I can find sample of simulation which contain  300 000 - 1000 000 elements ? 

![wydajnosc](https://user-images.githubusercontent.com/119892611/209226695-49f6aa01-e648-48b2-9689-634b239b7b9e.jpg)

**Comment by laurent-altr** [comment]
Hi @CinekPoland, 
A couple of remarks:
- With OpenRadioss we usually measure the speed-up by computer node, and not by core. When you go from 2 to 16 cores on the same machine, you have not multiplied the resources by 8. For instance, it is very much possible that cores are sharing some cache memory and memory bandwidth, that are critical for OpenRadioss. 
- Sph implementations known to have a limited scalability with OpenRadioss. 

I'll let others (@PaulAltair, @MarianBulla ) reply on the other models you are looking for.

Thanks

**Reply by CinekPoland** [reply]
Thank you for your answer. Could you explain exactly what you mean "cores are sharing some cache memory and memory bandwidth, that are **critical** for OpenRadioss." 

So it is not good when cores sharing cache memory and bandwidth ?

**Reply by laurent-altr** [reply]
I mean that OpenRadioss is memory bound. 

For example, let's say that we have a CPU with two cores sharing the L2 cache. When you are running on 1 core you are already using all the L2 available.  In that condition, you cannot expect a 2x speedup going from one to two cores.

**Reply by CinekPoland** [reply]
Do you think better scalability could be gained if  each processor core is assigned a fixed value of cache and ram ? 

I thought that if you set, for example, 16 cores to perform a given task, then each core will be assigned a specific value of the amount of cache memory, and not that one core takes up the entire cache value.
